packages/parrot-fetch/src/index.js

```diff
@@ -15,12 +15,20 @@
 import ParrotFetch from './ParrotFetch';
 
 /* istanbul ignore next */
-const context = typeof global === 'undefined' ? self : global; // eslint-disable-line no-restricted-globals
-const contextFetch = context.fetch.bind(context);
+const globalContext = typeof global === 'undefined' ? self : global; // eslint-disable-line no-restricted-globals
+const contextFetch = globalContext.fetch.bind(globalContext);
 export const PARROT_STATE = 'PARROT_STATE';
 
-export default function init(scenarios) {
-  const parrotFetch = new ParrotFetch(scenarios, contextFetch);
-  context.fetch = parrotFetch.resolve;
+export default function init(scenarios, contextParam = globalContext) {
+  const context = contextParam;
+  let parrotFetch;
+  // Option to mock a fetch client in a redux thunk object by attaching it to the context
```

Comment:
So are we coupling this to redux? Or do we just need to clean up this comment to not assume redux. Really any instance of fetch inside `global.thunks` will be used right?

Why `thunks`?

Reply:
it's not necessarily coupled to redux. you could pass in any object that has a fetch function in it (like the one app thunk object) and mock that fetch.

in my specific case, i'm passing in the redux thunk object, and mocking that fetchClient.

i could change the naming here to not include thunk and redux, because it would work without redux/thunk.

Reply:
i could probably also bind the fetchClient to the context object, and then only pass in the fetchClient function without the entire context object. but then it would require others who use this functionality to know that they need to do that bind too.

Reply:
yeah let's do that! And definitely add something about this in the README.